RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/RegistrationModels.swift

```diff
@@ -0,0 +1,197 @@
+// 
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// The parameters used for registration requests.
+struct RegistrationParameters: Codable {
+    /// Authentication parameters
+    var auth: AuthenticationParameters?
+    
+    /// The account username
+    var username: String?
+    
+    /// The account password
+    var password: String?
```

Comment:
This is how the parameters are defined in the spec for [/register](https://spec.matrix.org/v1.2/client-server-api/#post_matrixclientv3register). Depending on the step in the flow, a different set of parameters could be sent to the endpoint, e.g. [`RegistrationWizard.acceptTerms()`](https://github.com/vector-im/element-ios/pull/6056/files#diff-761629a5e9a91a9e9a504c9c2f84c933556aaa71166c0e67991eda94c52fcbedR109).

It would be possible to make stricter structs for each request and update `performRegistrationRequest` to accept any `Codable` type for the parameters, but I think that would only be pushing the problem further one step further down the line in reality.

Reply:
Hm I see, I did not realize this was used again in the interactive authentication flow, having them optional makes sense for this PR.

That being said it would still be nicer to create the stricted structs for each possible flow step, as it reduces the ambiguity in our type system, but that requires more refactors beyond the scope of this PR.

Reply:
Ok makes sense to me, added to the list 👍